### PR TITLE
Bugfix for issue 8458

### PR DESCRIPTION
**Pull request by subagonsouth**
Modified test TestColumnFunctions.test_coldefs_init_from_array for coverage of datatypes of little and big endianess

Fixed bug in ColDefs._init_from_array to cover all possible unsigned numpy datatypes

### CODE REVIEW

**Comment by pllim** [comment]
Please move milestone to 2.0.13. Sorry for any inconvenience caused.

**Comment by bsipocz** [comment]
Thank you @subagonsouth for the prompt fix, and @olebole for reporting the issue and confirming the fix works!

**Comment by olebole** [comment]
@subagonsouth big thanks for the quick fix!! With your patch, astropy [succeeds its build time tests on all official architctures](https://buildd.debian.org/status/logs.php?pkg=astropy&ver=3.1.2-2).

File: astropy/io/fits/column.py [hunk 1]
```diff
@@ -1423,12 +1423,13 @@ def _init_from_array(self, array):
...
-                bzero = np.uint32(2**31)
-            elif 'K' in format and ftype.base == np.dtype('uint64'):
-                bzero = np.uint64(2**63)
+            if ftype.base.kind == 'u':
+                if 'I' in format:
+                    bzero = np.uint16(2**15)
+                elif 'J' in format:
+                    bzero = np.uint32(2**31)
+                elif 'K' in format:
+                    bzero = np.uint64(2**63)
```
Comment (saimn) [inline astropy/io/fits/column.py]: Sorry for not replying sooner, I wanted to check the meaning of removing the itemsize comparison. The format letters are computer above (``from_recformat``) from the dtype, so it is actually a good things to remove this! Thanks @subagonsouth .

**Pull request merged**